thriftclient/client.go

```diff
@@ -0,0 +1,64 @@
+package thriftclient
+
+import (
+	"context"
+
+	"github.com/apache/thrift/lib/go/thrift"
+
+	"github.com/reddit/baseplate.go/log"
+	"github.com/reddit/baseplate.go/tracing"
+)
+
+// MonitoredClient implements the thrift.TClient interface and can be used to
+// wrap thrift calls in a client span.
+//
+// Calls will only be wrapped in client spans if the given context has a parent
+// span (either a server or "active" local/client span) set on it.  If no such
+// span is set, then the call will still proceed, it will just not be monitored
+// by a client span.
+type MonitoredClient struct {
+	client thrift.TClient
+}
+
+// NewMonitoredClientFromFactory returns a pointer to a new MonitoredClient that
+// a new TStandardClient using the provided transport and protocol factory.
+func NewMonitoredClientFromFactory(transport thrift.TTransport, factory thrift.TProtocolFactory) *MonitoredClient {
+	return NewMonitoredClient(thrift.NewTStandardClient(
+		factory.GetProtocol(transport),
+		factory.GetProtocol(transport),
+	))
+}
+
+// NewMonitoredClient returns a pointer to a new MonitoredClient that wraps the
+// given thrift.TClient calls in thrift client spans.
+func NewMonitoredClient(c thrift.TClient) *MonitoredClient {
+	return &MonitoredClient{client: c}
+}
+
+// Call wraps the underlying thrift.TClient.Call method with a client span.
+func (c MonitoredClient) Call(ctx context.Context, method string, args, result thrift.TStruct) error {
+	var span *tracing.Span
+	if span = tracing.GetActiveSpan(ctx); span == nil {
+		span = tracing.GetServerSpan(ctx)
+	}
+	var child *tracing.Span
+	var err error
+	if span != nil {
+		ctx, child = span.ChildAndThriftContext(ctx, method)
+	}
+	defer func() {
+		if child != nil {
+			if stopErr := child.Stop(ctx, err); stopErr != nil {
```

Comment:
What I mean is that at function signature you use named return (`(err error)` instead of just `error`), then you no longer need to define it in L45, and you can combine L56 and L57 into `return c.client.Call(ctx, method, args, result)`, and it still works in this defer'd block.